modules/system/assets/js/snowboard/extras/DataConfig.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Data configuration provider.
+ *
+ * Provides a mechanism for passing configuration data through an element's data attributes. This
+ * is generally used for widgets or UI interactions to configure them.
+ *
+ * @copyright 2022 Winter.
+ * @author Ben Thomson <git@alfreido.com>
+ */
+export default class DataConfig extends Snowboard.PluginBase {
```

Comment:
If nothing is currently using it I'd prefer that we don't include it for now.

Reply:
Have a read of the docs when they're done, and let me know if you still feel that way ;)